src/include/hash_ring.h

```diff
@@ -0,0 +1,76 @@
+#ifndef __HASH_RING_H__
```

Comment:
We should decide what style guide we want to follow. [Google's C++ Style Guide](https://google.github.io/styleguide/cppguide.html) says that [define guards](https://google.github.io/styleguide/cppguide.html#The__define_Guard) should have the form `FOO_BAR_BAZ_H` for file `foo/bar/baz.h`. I don't know if anyone else has strong opinions on what we should use, but we're inconsistent right now.

Reply:
Follow-up on this is in #61.